Crypto/Number/F2m.hs

```diff
@@ -209,3 +210,16 @@ divF2m
     -- ^ Quotient
 divF2m fx n1 n2 = mulF2m fx n1 <$> invF2m fx n2
 {-# INLINE divF2m #-}
+
+traceF2m :: BinaryPolynomial -> Integer -> Integer
+traceF2m fx = foldr addF2m 0 . take (log2 fx) . iterate (squareF2m fx)
```

Comment:
I believe that a pair of `take` and `iterate` should be replaced with `replicate`.

Reply:
Here we are defining a list of iterated squares:
```hs
[fx, squareF2m fx, squareF2m (squareF2m fx), squareF2m (squareF2m (squareF2m fx)), ...]
```
And then we take the sum (`foldr addF2m 0`) of a prefix of that list. I do not know how to do this with `replicate`, since that would just give us the last item of that prefix.

Reply:
My bad. I took it as `repeat`.